### PR TITLE
Hallway autonomous

### PR DESCRIPTION
**Pull request by ManiacMoustache**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Steps to Test
This was tested in an autonomous mode included with this change.
Uses navx + drivetrain encoders

## Changelog
We need to know how to use autonomous components to travel around the field accurately.

## What?
Added a new autonomous mode and made changes to turntoangle to make it better.

## Why?
We need to know how to use these components.

## Automation
#15

### CODE REVIEW

**Review by hell0w0rld2 — DISMISSED** [review]

**Review by hell0w0rld2 — DISMISSED** [review]
👍

File: autonomous/autonomous.py [hunk 1]
```diff
@@ -67,3 +70,114 @@ def engage_shooter(self):
...
+
+    @state(first = True)
+    def drive_forwards(self):
+        """Drives the bot forwards for 5 feet"""
+        self.driveTrain.setup()
+        self.driveTrain.resetDistTraveled()
+        self.goToDist.engage()
+        self.goToDist.start(-60)
+        self.driveTrain.execute()
+        self.next_state("stoprunning")
```
Comment (ChrisHirsch) [inline autonomous/autonomous.py:103]: stoprunning vs stoprunning2

File: autonomous/autonomous.py [hunk 2]
```diff
@@ -67,3 +70,114 @@ def engage_shooter(self):
...
+
+    @state
+    def turn(self):
+        """One method that completes all turns"""
+        turn1 = -90
+        turn2 = 180
+        turn3 = -90
+        self.turnToAngle.engage()
+        self.CheckAngleFirstCall = True
+        if self.TurnsCompleted == 0:
```
Comment (ChrisHirsch) [inline autonomous/autonomous.py:126]: Consider breaking these into their own states next time. So Turn1 is a state...and you continue in Turn1 until the turn is complete...once completed, self.next_state("turn2"),e tc

Comment (ChrisHirsch) [inline autonomous/autonomous.py:126]: Consider breaking these into their own states next time. So Turn1 is a state...and you continue in Turn1 until the turn is complete...once completed, self.next_state("turn2"),etc

File: autonomous/autonomous.py [hunk 3]
```diff
@@ -67,3 +70,114 @@ def engage_shooter(self):
...
+        elif self.TurnsCompleted == 1:
+            log.info("States Sencond turn")
+            self.turnToAngle.setAngle(turn2)
+            self.next_state("check_angle")
+        elif self.TurnsCompleted == 2:
+            self.turnToAngle.setAngle(turn3)
+            self.next_state("check_angle")
+
+    @state
+    def check_angle(self):
```
Comment (ChrisHirsch) [inline autonomous/autonomous.py:139]: consider this state as more of a methond that the Turn1 stage would do while checkAngle() != desired angle

**Review by ChrisHirsch — APPROVED** [review]
Make sure to review the comments above for when we do this for real

File: autonomous/autonomous.py (hunk 3 above)
Comment (ManiacMoustache) [reply autonomous/autonomous.py:141]: I think that this state is still a waiting state - so it isn't just a method, it actively does nothing. "turn" is just a signpost - it redirects turnToAngle while "check_angle" waits for it to complete.

**Review by hell0w0rld2 — APPROVED** [review]
👍

**Pull request closed without merging**